### DISCUSSION TITLE
qexo-friends支持hexo-circle-of-friends的css选择器抓取规则吗？

### DISCUSSION THREAD

**Discussion by ning0818** [Qexo/Qexo#322] [Ideas]
https://fcircle-doc.yyyzyyyz.cn/#/developmentdoc?id=%E5%8F%8B%E9%93%BE%E9%A1%B5%E9%80%82%E9%85%8D

![image](https://user-images.githubusercontent.com/117955401/219850413-a361e264-e086-4e6c-9f61-93136344a998.png)

**Comment by ning0818** [comment]
@am-abudu

**Comment by ning0818** [comment]
@Fgaoxing

**Comment by Fgaoxing** [comment]
无效 #95

**Reply by ning0818** [reply]
。。。

**Comment by am-abudu** [comment]
无法支持，因为本项目的原理是动态加载html页面。我不认为友联朋友圈支持

**Reply by Fgaoxing** [reply]
其实可以做，只是成本太高，超过本身的意义

**Reply by ning0818** [reply]
噢噢噢哦哦哦哦哦哦哦哦哦哦哦哦哦哦哦哦哦哦好吧

**Reply by Bytelv** [reply]
我前段时间就已经将这个问题给研究完了，目前是可以兼容的

**Comment by ning0818** [comment]
哦哦哦哦哦哦哦哦哦哦哦哦哦哦哦哦哦哦好吧

**Comment by Fgaoxing** [comment]
@ning0818 后续我可能会研究研究看看有没有可能，或者做个兼容的方案

**Reply by Bytelv** [reply]
我想你也许会需要我之前写的一篇文章：https://blog.lvbyte.top/Hexo+Stellar+Qexo/#%E6%B7%BB%E5%8A%A0Qexo%E5%8F%8B%E9%93%BE%E5%88%B0-%E6%9C%8B%E5%8F%8B%E5%9C%88
我利用python将两者的数据进行转换：
import json, requests 

url = requests.get("https://你的Qexo链接/pub/friends/")  #获取友链
text = url.text   #解析json数据
data = json.loads(text)     #获取data数据
data2 = data['data'] #data2为进入json data类中

#以上代码为解析Qexo友链json数据为python数据

#Friends_circle为friends字典   Friends_links为friends列表，其中应有len(data2)个嵌套的Friends列表   Friends列表为Qexo友链
#len(data2)为友链数量

Friends_circle={}  #定义列表和字典
Friends_links=[]
Friends=[]

num=len(data2)
count=0
for _ in range(num):  #重复输入友链数据至Friend_circle字典中
    Friends=[1,2,3]  #初始化Friends列表
    Friends[0]=data2[count]['name']   #Friends列表数据注入
    Friends[1]=data2[count]['url']
    Friends[2]=data2[count]['image']   
    Friends_links.append(Friends)
    Friends_circle ={"friends":  Friends_links   }
    Friends=[1,2,3]
    count=count+1  #进位

print(Friends_circle)
Friends_json=json.dumps(Friends_circle)
print("Json数据为：",Friends_json)
with open('friends.json', 'w') as f:
    f.write(Friends_json)
    f.close()

----------------------------------------
你可以试一下，看一下我写的文章。
我目前是成功抓取qexo的友链了：
看我朋友圈：https://blog.lvbyte.top/friend/rss/

**Reply by ning0818** [reply]
EN
